model/edx.go

```diff
@@ -67,6 +67,45 @@ func (e *EdxLMS) GetOAuthConsumerSecret() string {
 	return e.OAuthConsumerSecret
 }
 
+func (e *EdxLMS) GetTeams() map[string]string {
+	return e.Teams
+}
+
+func (e *EdxLMS) CheckIfActiveLMS(currentTeamName string) bool {
+	// For every property of Teams in this learning management service (LMS) object (passed as e)
```

Comment:
Sorry to keep adding review comments! 

Looking at this function again, I think I didn't think about what it was doing the first time round (another illustration of the importance of "why" comments!).

I _think_ what this function is doing is detecting whether `currentTeamName` matches any of the names of the teams stored in this EdxLMS structure. I _think_ the idea is that, if it does, then the passed team name must correspond to an "active" team. 

If that's so, I think it might be preferable to remove the `GetTeams()` function. It has a naming conflict with `GetTeams([]*model.Team, *model.AppError)` in `app/plugin_api.go`, and you're only using it here on line 78, when you could equally well write e.Teams (since you're still in a EdxLMS method). 

The comment also seems misleading or confusing to me. It is a bit tricky for me to tell what the `e.Teams` map is actually storing, but I think it maps context Ids to team names? If this is so, it might be clearer to say that we're going to iterate over all the names of the teams associated with this LMS. If not, it might be clearer to document what exactly that mapping is (properties is kinda vague).

Reply:
Just added more/revised documentation this function, let me know what you think. Also removed GetTeams, nice catch. I was using GetTeams from outside this package before, but now it makes sense just to use e.Teams.

Reply:
No worries on the comments, let's get this thing right! (Also, I'm learning how to write better code)

Reply:
This is definitely clearer, but there's kinda of a lot of "what" vs. "why" in the comments. Maybe:

//This function determines whether this LMS object (e) is active.
//An LMS is active if one of the team names it contains matches the team name of the current user.
//The team name of the current user is passed as a string, and compared to the names of all the teams stored in this LMS.

Reply:
I like it, but changed it slightly to:

`// This function determines whether this LMS object (e) is active
// An LMS is active if one of the team names it contains matches the team name of the currently selected MM team
// The team name of the currently selected MM team is passed as a string (currentTeamName), and compared to the names of all the teams stored in this LMS`

What do you think?

Reply:
I renamed this function to `IsLMSForTeam` and updated the function comment and implementation somewhat and pushed it for Brec a little while ago (I DM'd Bred so he knows about it) to look over, which may make this comment thread obsolete. (I also updated some of the rest of the LMS method changes)